### PR TITLE
Refactor public lists authentication logic

### PR DESCRIPTION
**Pull request by phamann**
We weren't previously taking into account the default fallback to `FT-Session` tokens.

### CODE REVIEW

File: server/middleware/auth/get-session.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+			headers: headers
+		})
+		.then(response => {
+			if (response.status === 404) {
+				throw 401;
+			}
+			return response;
+		})
+		.then(fetchres.json);
+	}	else {
```
Comment (laurieboyes) [inline server/middleware/auth/get-session.js]: weird formating :pendantic_cat:

File: server/middleware/auth/list.js [hunk 2]
```diff
@@ -23,16 +24,24 @@ module.exports = (req, res, next) => {
...
-
-			next();
+			let uuid = req.get('FT-User-UUID');
+			let userPromise = uuid ? Promise.resolve({uuid}) : getSession(req);
 
+			userPromise.then(user => {
+					if(listOwner === user.uuid) {
+						res.locals.userId = user.uuid;
+						next();
+						return;
```
Comment (laurieboyes) [inline server/middleware/auth/list.js]: unnecessary return statement :pedantic_owl:

File: server/middleware/auth/list.js [hunk 3]
```diff
@@ -23,16 +24,24 @@ module.exports = (req, res, next) => {
...
+					} else {
+						return Promise.reject();
+					}
+				})
+				.catch(() => {
+					if(isPublic) { next(); return; }
+					let msg = messages.UNAUTHORIZED;
+					res.status(msg.statusCode).send(msg.error);
+					return;
```
Comment (laurieboyes) [inline server/middleware/auth/list.js]: unnecessary return statement

File: server/middleware/auth/list.js [hunk 4]
```diff
@@ -23,16 +24,24 @@ module.exports = (req, res, next) => {
...
-				res.status(msg.statusCode).send(msg.error);
-				return;
-			}
-
-			next();
+			let uuid = req.get('FT-User-UUID');
+			let userPromise = uuid ? Promise.resolve({uuid}) : getSession(req);
 
+			userPromise.then(user => {
+					if(listOwner === user.uuid) {
```
Comment (keirog) [inline server/middleware/auth/list.js]: Inconsistent space police :rotating_light: `if(` & `if (`

File: server/middleware/auth/list.js [hunk 5]
```diff
@@ -23,16 +24,24 @@ module.exports = (req, res, next) => {
...
+						res.locals.userId = user.uuid;
+						next();
```
Comment (keirog) [inline server/middleware/auth/list.js]: Opportunity for `return next();`

File: server/middleware/auth/list.js [hunk 6]
```diff
@@ -23,16 +24,24 @@ module.exports = (req, res, next) => {
...
+						return;
+					} else {
+						return Promise.reject();
+					}
+				})
+				.catch(() => {
+					if(isPublic) { next(); return; }
```
Comment (keirog) [inline server/middleware/auth/list.js]: Opportunity for `return next();`

**Pull request merged**